app/services/my_anime_list_sync_service.rb

```diff
@@ -0,0 +1,203 @@
+  MINE = '?mine=1'.freeze
+
+  attr_reader :le, :method
+
+  def initialize(le, method)
+    @le = le
+    @method = method
+  end
+
+  def execute_method
+    # anime or manga
+    media_type = le.media_type.underscore
+    # convert kitsu data -> mal data
+    mal_media = le.media.mappings.find_by(site: "myanimelist/#{media_type}")
+    # user mal profile information
+    profile = LinkedProfile.find_by(
+      user_id: le.user_id,
+      url: 'myanimelist'
+    )
+    mal_media_id = mal_media.external_id
+
+    case method
+    when 'delete'
+      delete(
+        "#{animelist}/#{mal_media_type}/#{mal_media_id}",
+        profile
+      )
+    else
+      # find the anime or manga
+      get("#{media_type}/#{mal_media_id}#{MINE}", profile) do |response|
+        # need to check watched_status OR read_status
+        if media_type == 'anime'
+          if response['watched_status']
+            put(
+              "animelist/#{media_type}/#{mal_media_id}",
+              profile,
+              {
+                status: format_status(le.status),
+                episodes: le.progress,
+                score: format_score(le.rating),
+                rewatch_count: le.reconsume_count
+              }
+            )
+          else
+            post(
+              "animelist/#{media_type}",
+              profile,
+              {
+                anime_id: mal_media_id,
+                status: format_status(le.status),
+                episodes: le.progress,
+                score: format_score(le.rating)
+              }
+            )
+          end
+        else
+          # manga
+          if response['read_status']
+            put(
+              "mangalist/#{media_type}/#{mal_media_id}",
+              profile,
+              {
+                status: format_status(le.status),
+                chapters: le.progress,
+                score: format_score(le.rating),
+                reread_count: le.reconsume_count
+              }
+            )
+          else
+            post(
+              "mangalist/#{media_type}",
+              profile,
+              {
+                manga_id: mal_media_id,
+                status: format_status(le.status),
+                chapters: le.progress,
+                score: format_score(le.rating)
+              }
+            )
+          end
+        end
+      end
+
+      # if block is not executed because
+      # response.code was not a 200
+      # TODO: check timing
+      # return nil
+    end
+  end
+
+  def format_status(status)
+    # change our status -> mal status
+    case status
+    when 'current' then 1 # watching/reading
+    when 'planned' then 6 # plan to watch/plan to read
+    when 'completed' then 2 # completed
+    when 'on_hold' then 3 # on hold
+    when 'dropped' then 4 # dropped
+    end
+  end
+
+  # if you send no score in
+  # ie: &score&anythingelse
+  # it will not set the score
+  def format_score(score)
+    return nil if score.nil?
+
+    (score * 2).floor
```

Comment:
Might be cleaner to just say

```ruby
def format_score(score)
  (score * 2).to_i if score
end
```

There's no reason to floor it, all our scores are .0 or .5, which when multiplied by two guarantees a .0